include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp

```diff
@@ -262,9 +260,10 @@ class alignment_executor_two_way
             current_chunk_size = std::ranges::distance(current_chunk);
             assert(in_avail() >= current_chunk_size);
 
-            exec_handler.execute(kernel, std::move(current_chunk), [write_to = gptr] (auto && alignment_results)
+            exec_handler.execute(kernel, std::move(current_chunk), [write_to = gptr] (auto && alignment_result) mutable
```

Comment:
Why does this need to be mutable now? (or maybe why didn't is need to be mutable before)

Reply:
All captured arguments are const by default unless you declare the lambda mutable. Before I had a const pointer but didn't care because I dereferenced it and since const on pointers does not propgate I could write to the memory that write_to was pointing to. The current change implies that the algorithm calls the delgate once for every computed alignment result instead of collecting them in a vector and returning the vector. So I know have to make sure that the lambda updates the `write_to` position, which is only possible if it is not `const`.